app/apps/views.js

```diff
@@ -162,13 +108,38 @@ exports.app_edit = [
       .all([app_request, buckets_request, users_request])
       .then(function(responses) {
         let [app_response, buckets_response, users_response] = responses;
+        const all_buckets = buckets_response.results;
+        const bucket_ids_associated = app_response.apps3buckets.map(as => as.s3bucket.id);
+        const buckets_other = all_buckets.filter(bucket => {
+          return bucket_ids_associated.indexOf(bucket.id) < 0;
+        });
```

Comment:
I would probably move some of this logic out once I start to add the other views and I have a better understanding of the common bits.

I agree this would probably end up in an utility function to "get the things I'm not already associated with" or something like that.

Reply:
It's not about moving out common bits as much as self-documenting code.